packages/cli/src/utils/errors.ts

```diff
@@ -1,54 +1,47 @@
-// Based on hardhat's errors and errors-list (MIT)
-// https://github.com/NomicFoundation/hardhat/tree/main/packages/hardhat-core
-
-export interface ErrorDescriptor {
-  message: (...args: string[]) => string;
-  description: string;
+import chalk from "chalk";
+import { ZodError } from "zod";
+import { fromZodError, ValidationError } from "zod-validation-error";
+
+// Wrapper with preset styles, only requires a `prefix`
+export function fromZodErrorCustom(error: ZodError, prefix: string) {
+  return fromZodError(error, {
+    prefix: chalk.red(prefix),
+    prefixSeparator: "\n- ",
+    issueSeparator: "\n- ",
+  });
 }
 
-export class CustomError extends Error {
-  constructor(message: string, public readonly parent?: Error) {
-    super(message);
-
-    this.name = this.constructor.name;
-  }
+export class NotInsideProjectError extends Error {
+  name = "NotInsideProjectError";
+  message = "You are not inside a MUD project";
 }
 
-export class MUDError extends CustomError {
-  public static isMUDError(other: any): other is MUDError {
-    return other !== undefined && other !== null && other._isMUDError === true;
-  }
-
-  public readonly errorDescriptor: ErrorDescriptor;
-
-  private readonly _isMUDError: boolean;
-
-  constructor(errorDescriptor: ErrorDescriptor, messageArguments: string[] = [], parentError?: Error) {
-    super(errorDescriptor.message(...messageArguments), parentError);
-
-    this.errorDescriptor = errorDescriptor;
+export class NotESMConfigError extends Error {
+  name = "NotESMConfigError";
+  message = "MUD config must be an ES module";
+}
 
-    this._isMUDError = true;
+export function logError(error: Error) {
+  if (error instanceof ValidationError) {
+    console.log(chalk.redBright(error.message));
+  } else if (error instanceof ZodError) {
+    // TODO currently this error shouldn't happen, use `fromZodErrorCustom`
+    const validationError = fromZodError(error, {
+      prefixSeparator: "\n- ",
+      issueSeparator: "\n- ",
+    });
+    console.log(chalk.redBright(validationError.message));
```

Comment:
> Or should we add a console log saying "If you see this log, please report at ..."?

Strongly considered this actually, but left as-is because this is still WIP and I might end up needing this error or changing this whole method or something